src/amo/pages/Home/index.js

```diff
@@ -274,7 +274,7 @@ export class HomeBase extends React.Component {
           {isDesktopSite ? (
             <HeroRecommendation
               errorHandler={errorHandler}
-              shelfData={heroShelves && heroShelves.primary}
+              shelfData={heroShelves ? heroShelves.primary : undefined}
```

Comment:
my logic in javascript is rusty but `heroShelves && heroShelves.primary` would be `null` if `heroShelves` is `null`, right?  You just made changes above in `HeroRecommendation` to explicitly handle when `shelfData` is `null`, so why do we need it to be `undefined`?

Reply:
Yes, that's correct (your logic). The issue is, when `heroshelves` is `null`, then `null` would be passed as `shelfData`, but the `HeroRecommendation` component now interprets that as meaning that `heroShelves.primary` was `null`, which means something different.

Essentially, when `heroshelves` is `null` or really any falsey value, the `HeroRecommendation` component is still waiting for the data, so it should display itself in a loading state. This is old logic that we need to keep. The new use case is when `heroshelves` is not falsey *and* `heroShelves.primary` is `null`, which indicates that we did get an API response, but there is no data for the primary hero. In that case we need to hide the entire `HeroRecommendation` component (as opposed to showing it in a loading state), and this is communicated by passing `null` as `shelfData` (because `heroShelves.primary` is `null`). I suppose communicating these two cases to the `HeroRecommendation` component could be done in a number of different ways, but the simplest thing seemed to be to keep most of the existing logic, and just make sure that `shelfData` is `undefined` when we are still waiting for a result from the API.

Does that make sense?

Reply:
it's not a particularly obvious way of specifying the difference between loading and the data existing but I suppose it makes intuitive sense that undefined means we don't know about the data yet.

Reply:
Yeah. I didn't want to add a new property for this. I think it works.